hooks/syslog/syslog.go

```diff
@@ -4,50 +4,96 @@ package syslog
 
 import (
 	"fmt"
-	"log/syslog"
+	//"log/syslog"
+	"github.com/GolangResources/syslog/syslog"
 	"os"
+	"time"
 
 	"github.com/sirupsen/logrus"
+	sg "log/syslog"
+)
+
+const (
+	NUM_SYSLOG_WORKERS = 8
+	CHANNEL_DEPTH = 8192
 )
 
 // SyslogHook to send logs via syslog.
 type SyslogHook struct {
-	Writer        *syslog.Writer
-	SyslogNetwork string
-	SyslogRaddr   string
+	Writer		*syslog.Writer
+	SyslogNetwork	string
+	SyslogRaddr	string
+	Tag		string
+	syschan		chan *SEntry
+}
+
+type SEntry struct {
+	line		string
+	level		logrus.Level
 }
 
 // Creates a hook to be added to an instance of logger. This is called with
 // `hook, err := NewSyslogHook("udp", "localhost:514", syslog.LOG_DEBUG, "")`
 // `if err == nil { log.Hooks.Add(hook) }`
-func NewSyslogHook(network, raddr string, priority syslog.Priority, tag string) (*SyslogHook, error) {
-	w, err := syslog.Dial(network, raddr, priority, tag)
-	return &SyslogHook{w, network, raddr}, err
+func NewSyslogHook(network, raddr string, priority sg.Priority, tag string) (*SyslogHook, error) {
+	w, err := syslog.Dial("ctcp" , raddr, priority, nil)
+	if err != nil {
+		return nil, err
+	}
+	hook := &SyslogHook{
+		w,
+		network,
+		raddr,
+		tag,
+		make(chan *SEntry, CHANNEL_DEPTH),
```

Comment:
Could the channel creation be configurable ?

We need to be able to keep a compatible behaviour with the former version.